submit-api/src/Cardano/TxSubmit/CLI/Parsers.hs

```diff
@@ -103,13 +103,16 @@ pProtocol =
   <|>
     (  Opt.flag' ()
         (  Opt.long "cardano-mode"
-        <> Opt.help "For talking to a node running in full Cardano mode."
+        <> Opt.help "For talking to a node running in full Cardano mode (default)."
         )
     *> pCardano
     )
   <|>
-    -- Default to the Shelley protocol for now, due to the testnet.
-    pure ShelleyProtocol
+    -- Default to the Cardano protocol.
+    pure
+      (CardanoProtocol
+        (EpochSlots defaultByronEpochSlots)
+        (SecurityParam defaultSecurityParam))
```

Comment:
With `optparse-applicative`, instead of doing `Opt.option (...<>...) <|> pure Something` I think it's better use `Opt.value` like so:

```
Opt.flag' ()
        (  Opt.long "cardano-mode"
        <> Opt.help "For talking to a node running in full Cardano mode (default)."
        <> Opt.value (CardanoProtocol ...)
        )
```

It's easier to see that the default value's connected to the option.

Reply:
With the way I've got this `Parser` setup, it's a bit tricky to do that here. 

I think that, in the near future, I'll follow your advice from Slack and go the subcommand route instead since this probably isn't the nicest way to do it.